src/main.rs

```diff
@@ -67,7 +67,22 @@ fn validate_simulation() {
 
 /// Simulate the rabbit population
 fn simulate(starting_rabbits: i128) -> i32 {
-    todo!()
+    let mut rabbits = starting_rabbits;
+    let mut days = 0;
+
+    loop {
+        if rabbits == 1{
+            return days;
+            //break;
+        }
+        if rabbits%2 == 0 {
+            rabbits = rabbits/2;
```

Comment:
One thing I saw from some other students (that I didn't think to do myself) was to do `rabbits /= 2;` on this line. Kind of neat!